ludwig/data/sampler.py

```diff
@@ -0,0 +1,64 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2020 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import math
+
+import numpy as np
+
+
+class DistributedSampler:
+    """Adapted from `torch.utils.data.distributed.DistributedSampler`."""
+    def __init__(self, dataset, shuffle=True, seed=0, horovod=None):
+        self.dataset = dataset
```

Comment:
It seems like we need the dataset just for using its length, maybe we can pass a dataset_length parameter in the signature instead?

Reply:
Changed to `dataset_size`.